### ISSUE TITLE
Can't open files en SFTP after updating

### ISSUE THREAD

**Issue by maniattico** [h3imdall/ftp-remote-edit#257]
Hello.

I updated the version to 0.13.3 and for that moment I can't open files in sftp server.

Ctrl+Space doesn't works. I open with Package menu. I conect to the sftp server and list the files. However the files keep loading until I close atom.

Atom 1.34.0

Thanks.

**Comment by cschindl** [comment]
Hi @maniattico, 

You're right, Ctrl+Space does not currently work if you have a file open. I will correct that. When opening files via SFTP I can't find any error. What is displayed in the remote transfer log? Is the file trying to download?

**Comment by mkismy** [comment]
Since `ctrl-space` is binded to language switch as a default Mac shortcut, the changes in 0.13.3 is also causing confilict.
I've been using `shift-cmd-space` as a workaround.

**Comment by AL1nk2Th3Futur3** [comment]
I'm having the same issue. The file shows that it's attempting to download but nothing happens. The transfer log says that it's transferring but continues to hang. I've downgraded for the time being as a workaround.

**Comment by cschindl** [comment]
Can you tell me if in the transfer log the status changes from waiting to transfering and if the progress changes? I can't reproduce the error and need your help. I have added some additional debug messages. Can you please upgrade to the new version, enable debug mode and post the output here?

**Comment by AL1nk2Th3Futur3** [comment]
Something I thought of that might be useful to you is that I'm running Manjaro Architect Linux

The status immediately goes to transferring (except in the case where I try to download a second file alongside the first, that one will be waiting).

The progress doesn't change. 

Having never used the debug/dev mode before I'm not entirely sure if what I did was correct. I started atom with the --dev flag set, opened up the dev console (Ctrl+Shift+i > console) and then tried downloading a file. There was nothing logged to the console so I'm not certain if I did it correctly or not.

**Comment by cschindl** [comment]
To enable the debug mode please go to the atom settings and look for the package "ftp-remote-edit". Here you have to click on settings. Please activate the checkbox for debug mode. Then you can open the console via the menu "View" -> "Developer" -> "Toggle Developer Tools". After that please try to connect to the sftp server and download a file. The debug messages should be displayed in the console.

**Comment by AL1nk2Th3Futur3** [comment]
`connector:listDirectory /
tree-view.js? [sm]:334 connector:connect
tree-view.js? [sm]:334 sftp:connect
tree-view.js? [sm]:334 sftp:connect:ready
tree-view.js? [sm]:334 sftp:list
tree-view.js? [sm]:334 connector:disconnect
tree-view.js? [sm]:333 connector:listDirectory /home/
tree-view.js? [sm]:334 connector:connect
tree-view.js? [sm]:334 sftp:list
tree-view.js? [sm]:334 connector:disconnect
tree-view.js? [sm]:333 connector:listDirectory /home/user/
tree-view.js? [sm]:334 connector:connect
tree-view.js? [sm]:334 sftp:list
tree-view.js? [sm]:334 connector:disconnect
tree-view.js? [sm]:331 connector:downloadFile /home/user/.viminfo /tmp/83cec0b6/10.144.84.220/home/user/.viminfo
tree-view.js? [sm]:334 connector:connect
tree-view.js? [sm]:334 sftp:get
tree-view.js? [sm]:334 sftp:get:client.get:success
tree-view.js? [sm]:334 sftp:get:stream.pipe
tree-view.js? [sm]:334 sftp:get:file.open`

How's this?

The directories list fine, it's just downloading a file a that hangs

**Comment by cschindl** [comment]
Can you clone this repository into your atom package directory and initialize it with the console command "apm install"? Then I don't have to release a new version for every small change.  I've made a few changes to the SFTP module which you'll need to test once.

**Comment by AL1nk2Th3Futur3** [comment]
Looks to still be an issue. Continues to hang. Console doesn't log anything new either... Don't really have anymore information to provide...

**Comment by cschindl** [comment]
Is there possibly a possibility that you can set up a test account for me for a server, where I can test the whole one in detail?

**Comment by cschindl** [comment]
I've made a few more changes. Please pull from the repository and try again. Please also post the log from the console.

**Comment by AL1nk2Th3Futur3** [comment]
Sorry this took so long.


fs.statSyncNoException @ fs.js:958
tree-view.js? [sm]:333 connector:listDirectory /
tree-view.js? [sm]:334 connector:connect
tree-view.js? [sm]:334 sftp:connect
tree-view.js? [sm]:334 sftp:connect:ready
tree-view.js? [sm]:334 sftp:list
tree-view.js? [sm]:334 connector:disconnect
tree-view.js? [sm]:333 connector:listDirectory /home/
tree-view.js? [sm]:334 connector:connect
tree-view.js? [sm]:334 sftp:list
tree-view.js? [sm]:334 connector:disconnect
tree-view.js? [sm]:333 connector:listDirectory /home/user/
tree-view.js? [sm]:334 connector:connect
tree-view.js? [sm]:334 sftp:list
tree-view.js? [sm]:334 connector:disconnect
tree-view.js? [sm]:331 connector:downloadFile /home/user/.profile /tmp/83cec0b6/10.144.84.220/home/user/.profile
tree-view.js? [sm]:334 connector:connect
tree-view.js? [sm]:334 sftp:get
tree-view.js? [sm]:334 sftp:get:client.get:success
tree-view.js? [sm]:334 sftp:get:stream.pipe
tree-view.js? [sm]:334 sftp:get:file.open

New change is still a dud it seems. It must be something OS specific I'd think. What do you have in mind in terms of a server?

**Comment by cschindl** [comment]
Right now I don't know if it's the server or the os. With me I cannot adjust the error, because I have neither the server nor the os. Atom still uses different nodejs and electron versions for each os. So if I could get access to a server where the error occurs, I could try to fix it.

**Comment by AL1nk2Th3Futur3** [comment]
I've just tried it on a Windows machine. Worked flawlessly. The server I'm connecting to doesn't appear to be the issue. I'll mess around and see if I can find anything

**Comment by cschindl** [comment]
Can you please post your version here? Just execute the command "apm -v". I would be interested to know which nodejs version you are using.

**Comment by ssfrr** [comment]
I'm having this issue as well (also on Arch linux, though the standard Arch rather than Manjaro).

```
$ apm -v
apm  2.1.3
npm  6.8.0
node 11.10.0 x64
atom 1.34.0
python 2.7.15
git 2.20.1
```

Thanks for looking into this! I know remote debugging is slow and annoying.

I cloned the repo and opened in dev mode, and enabled debug mode in the package settings.

When atom first loads I get several thousand copies of the warning:
```
The 'lstatSyncNoException' function has been deprecated and marked for removal.
```
which looks like it's being used by the `fs-plus` library

I also get

```
[DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
```

but that looks like it's coming from Atom internals.
Here's the console output when I try to open a remote file:

```
connector:listDirectory /home/sfr/Dropbox/juliadev/
tree-view.js? [sm]:334 connector:connect
tree-view.js? [sm]:334 sftp:connect
tree-view.js? [sm]:334 sftp:connect:ready
tree-view.js? [sm]:334 sftp:list
tree-view.js? [sm]:334 connector:disconnect
[Violation] Forced reflow while executing JavaScript took 33ms
tree-view.js? [sm]:333 connector:listDirectory /home/sfr/Dropbox/juliadev/Ogg/
tree-view.js? [sm]:334 connector:connect
tree-view.js? [sm]:334 sftp:list
tree-view.js? [sm]:334 connector:disconnect
tree-view.js? [sm]:333 connector:listDirectory /home/sfr/Dropbox/juliadev/Ogg/test/
tree-view.js? [sm]:334 connector:connect
tree-view.js? [sm]:334 sftp:list
tree-view.js? [sm]:334 connector:disconnect
tree-view.js? [sm]:331 connector:downloadFile /home/sfr/Dropbox/juliadev/Ogg/test/runtests.jl /tmp/tmp.Km5zs7fRdZ/9651f1aa/eye.media.mit.edu/home/sfr/Dropbox/juliadev/Ogg/test/runtests.jl
tree-view.js? [sm]:334 connector:connect
tree-view.js? [sm]:334 sftp:get
tree-view.js? [sm]:334 sftp:get:client.get:success
tree-view.js? [sm]:334 sftp:get:stream.pipe
tree-view.js? [sm]:334 sftp:get:file.open
```

**Comment by AL1nk2Th3Futur3** [comment]
My Atom verion:

`apm  2.1.3
npm  6.7.0
node 11.8.0 x64
atom 1.34.0
python 2.7.15
git 2.20.1`

I got the same 'lstatSyncNoException' error as @ssfrr... Sorry, I should have included that. Still new to reporting bugs.

**Comment by ssfrr** [comment]
I just did a `git bisect` and it looks like 83246393 is the commit that introduced the issue.

**Comment by cschindl** [comment]
Your node version seems to be the cause of the error. All atom versions tested by me use max. version 9, but since version 10 the stream has changed a bit. I have now forced and modified the sftp library. Please pull the new version and check if this leads to success.

**Comment by ssfrr** [comment]
I pulled the latest commit (a7c977b) and it doesn't seem to solve the issue.

I tried downgrading `nodejs`, which also required downgrading `apm`, but then that broke Atom so I abandoned that plan.

**Comment by cschindl** [comment]
After you updated the repository, did you also update the plugin with "apm install" in the plugin directory?  Otherwise the dependencies are not up to date.  Atom does this whenever a new version is installed from the store.

**Comment by cschindl** [comment]
@ssfrr Have you updated the plugin again?

**Comment by ssfrr** [comment]
Sorry for the delay. Yes, I ran `apm install` after updating the git repo (it gave me an error otherwise).

**Comment by cschindl** [comment]
I was able to reproduce the error after installing arch linux on a vm. I adapted the fork of the sftp library. Because I don't know if "apm install" updates the library, please delete the "node_modules" subfolder in the plugin folder and then run "apm install". Please check if the bug is fixed for you.

**Comment by ssfrr** [comment]
:tada:

After removing the `node_modules` directory and re-running `apm install` the problem is solved on latest master (b15574b).

Thanks for digging into this!

**Comment by cschindl** [comment]
Changes are now included in the new release ([0.13.5](https://github.com/h3imdall/ftp-remote-edit/releases/tag/v0.13.5)).
